ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -0,0 +1,231 @@
+import Link from "next/link";
+import { useEffect, useMemo, useState } from "react";
+import { Button, Card, Col, Row, Table } from "react-bootstrap";
+import useSWR from "swr";
+import { useGrouparooModelContext } from "../../contexts/grouparooModel";
+import { ApiHook, UseApi } from "../../hooks/useApi";
+import { Actions, Models } from "../../utils/apiData";
+import { successHandler } from "../../utils/eventHandlers";
+import { grouparooUiEdition } from "../../utils/uiEdition";
+import StateBadge from "../badges/StateBadge";
+import ManagedCard from "../lib/ManagedCard";
+import SeparatedItems from "../lib/SeparatedItems";
+import LinkButton from "../LinkButton";
+import LoadingButton from "../LoadingButton";
+import AddSampleRecordModal from "./AddSampleRecordModal";
+
+interface Props {
+  execApi: ApiHook["execApi"];
+  properties: Models.PropertyType[];
+}
+
+interface RecordRow {
+  propertyKey: string;
+  propertyId: string;
+  propertyState: Models.PropertyType["state"];
+  value: string | number | boolean;
+}
+
+const isConfigUI = grouparooUiEdition() === "config";
+
+const SampleRecordCard: React.FC<Props> = ({ properties, execApi }) => {
+  const model = useGrouparooModelContext();
+
+  const [importing, setImporting] = useState(false);
+  const [addingRecord, setAddingRecord] = useState(false);
+  const [recordId, setRecordId] = useState(() => {
+    if (typeof window !== "undefined") {
+      return window.localStorage.getItem(`sampleRecord:${model.id}`);
+    }
+    return undefined;
+  });
+
+  const { data: record, mutate: mutateRecord } = useSWR(
+    ["sample-record", recordId],
+    () => {
+      if (recordId) {
+        return execApi<Actions.RecordView>("get", `/record/${recordId}`).then(
+          ({ record }) => record
+        );
+      }
+
+      return execApi<Actions.RecordsList>("get", "/records", {
+        limit: 1,
+        offset: 0,
+        modelId: model.id,
+        recordId,
+      }).then((data) => data?.records?.[0]);
+    }
+  );
+
+  useEffect(() => {
+    if (record) {
+      window.localStorage.setItem(`sampleRecord:${model.id}`, record.id);
+    }
```

Comment:
👍 local storage seems like the right way to store this.  I imagine setters and getters for sampleRecords will be abstracted once this is shared on more pages